### PR TITLE
Docs: add device limit guide

### PR DESCRIPTION
**Pull request by sixhobbits**
Depends on 

* https://github.com/ritza-co/fusionauth-device-limit-guide-simple
* https://github.com/ritza-co/fusionauth-device-limit-guide-friendly

being forked to fusionauth org

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206595519378128

### CODE REVIEW

File: astro/src/content/docs/extend/examples/device-limiting.mdx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+ * **Simpler implementation:** The user will simply be informed that they are logged in from too many devices and will be asked to log out of one of them manually, and then try logging in again. By logging out of one of the sessions, the refresh token for that session will be revoked by FusionAuth.
+
+ * **More user-friendly implementation:** The list of current logins will be displayed and the user will be able to select which session to end. The application will then call the FusionAuth API to end the session by [revoking the refresh token](/docs/apis/jwt#revoke-refresh-tokens) for the chosen session.
+
+For the second option to work, the user JWTs issued by FusionAuth must be configured to relatively short lifespans to prevent the user from staying logged in on the additional devices long after the corresponding refresh token has been revoked. 
+
+## The ChangeBank Application
+
+To make things a bit more concrete, let's use an example from the [quickStart guides](/docs/quickstarts/quickstart-javascript-express-web). The [ChangeBank](https://www.youtube.com/watch?v=CXDxNCzUspM) application is a simple banking application that allows users to convert dollars to coins.
```
Comment (vcampitelli) [inline astro/src/content/docs/extend/examples/device-limiting.mdx:49]:
```suggestion
To make things a bit more concrete, let's use an example from the [quickstart guides](/docs/quickstarts/quickstart-javascript-express-web). The [ChangeBank](https://www.youtube.com/watch?v=CXDxNCzUspM) application is a simple banking application that allows users to convert dollars to coins.
```

File: astro/src/content/docs/extend/examples/device-limiting.mdx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+To set up your FusionAuth application to issue refresh tokens, navigate to <strong>Applications -> Edit -> OAuth</strong> and select the <InlineField>Generate refresh tokens</InlineField> field if it isn't already selected. Scroll down to the <InlineField>Enabled Grants</InlineField> field and select the <InlineField>Refresh Token</InlineField> field.
+
+To set up your client application to request refresh tokens when logging in, add [the `offline_access` scope](/docs/lifecycle/authenticate-users/oauth/) in the `scope` parameter of the `/oauth2/authorize` request. If you are constructing the request manually, it should look something like the following.
+
+```javascript
+res.redirect(302, `${fusionAuthURL}/oauth2/authorize?client_id=${clientId}&response_type=code&scope=offline_access&redirect_uri=http://localhost:${port}/oauth-redirect&state=${userSessionCookie?.stateValue}&code_challenge=${userSessionCookie?.challenge}&code_challenge_method=S256`)
+```
+If you are using Passport.js, you can include the `offline_access` scope in the `scope` key of the options object passed to the `authenticate` method. It should look something like the following.
```
Comment (vcampitelli) [inline astro/src/content/docs/extend/examples/device-limiting.mdx:85]:
```suggestion
If you are using [Passport.js](https://www.passportjs.org/), you can include the `offline_access` scope in the `scope` key of the options object passed to the `authenticate` method. It should look something like the following.
```

File: astro/src/content/docs/extend/examples/device-limiting.mdx [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+## The User-Friendly Implementation
+
+The previous implementation has the advantage of code simplicity, but it does have a few problems. 
+
+* **Inconvenience:** The user has to manually log out of one of their devices before they can log in again. 
+
+* **The user is not informed which devices they are logged in on:** They will have to try to remember which of their devices they are logged in on. If they don't have physical access to one of their devices, they will be unable to log in. 
+
+* The failed login message is the same generic message used for all webhook errors:** Adding other webhooks to your application will result in the same error message being displayed, regardless of the type of webhook failing.
```
Comment (vcampitelli) [inline astro/src/content/docs/extend/examples/device-limiting.mdx:149]:
```suggestion
* **The failed login message is the same generic message used for all webhook errors:** Adding other webhooks to your application will result in the same error message being displayed, regardless of the type of webhook failing. 
```

File: astro/src/content/docs/extend/examples/device-limiting.mdx [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+
+<RemoteCode url={frontmatter.codeRootFriendly + "/complete-application/src/index.ts"} tags="device-limiting-maxcount" lang="typescript"/>
+
+The Handlebars template page should look similar to the following.
+
+<RemoteCode url={frontmatter.codeRootFriendly + "/complete-application/templates/device-limit.hbs"} lang="html"/>
+
+### Revoking A Chosen Session
+
+The web page posts the selected token Ids to the backend. You will need a route that accepts these token Ids and revokes the selected tokens using the DELETE method on the FusionAUth Refresh Token API.
```
Comment (vcampitelli) [inline astro/src/content/docs/extend/examples/device-limiting.mdx:208]:
```suggestion
The web page posts the selected token Ids to the backend. You will need a route that accepts these token Ids and revokes the selected tokens using the DELETE method on the FusionAuth Refresh Token API.
```

File: astro/src/content/docs/extend/examples/device-limiting.mdx [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+navcategory: developer
+section: extend
+subcategory: examples
+codeRootSimple: https://raw.githubusercontent.com/FusionAuth/fusionauth-device-limit-guide-simple/main
+codeRootFriendly: https://raw.githubusercontent.com/FusionAuth/fusionauth-device-limit-guide-friendly/main
+---
+import Aside from 'src/components/Aside.astro';
+import InlineField from 'src/components/InlineField.astro';
+import InlineUIElement from 'src/components/InlineUIElement.astro';
+import {RemoteCode, RemoteValue} from '@fusionauth/astro-components';
```
Comment (vcampitelli) [inline astro/src/content/docs/extend/examples/device-limiting.mdx:13]:
```suggestion
import {RemoteCode} from '@fusionauth/astro-components';
```

**Review by vcampitelli — CHANGES_REQUESTED** [review]
Just some small typos

**Comment by sixhobbits** [comment]
@vcampitelli - sorry about those. Let me know if any other feedback.

**Comment by mooreds** [comment]
@vcampitelli is this good to merge?

**Review by vcampitelli — APPROVED** [review]

**Pull request merged**